llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -2824,10 +2824,7 @@ SDValue SITargetLowering::LowerFormalArguments(
   SIMachineFunctionInfo *Info = MF.getInfo<SIMachineFunctionInfo>();
 
   if (Subtarget->isAmdHsaOS() && AMDGPU::isGraphics(CallConv)) {
-    DiagnosticInfoUnsupported NoGraphicsHSA(
-        Fn, "unsupported non-compute shaders with HSA", DL.getDebugLoc());
-    DAG.getContext()->diagnose(NoGraphicsHSA);
-    return DAG.getEntryNode();
+    report_fatal_error("unsupported non-compute shaders with HSA");
```

Comment:
This should not be a fatal error. You should adjust the return value to be the correct type by using the correct set of undef values

Reply:
What is the point? There will be no valid output anyways. Trying to validate even invalid output takes a lot of effort and will ityself be fragile because that code is rarely executed/tested.